servers/services/src/main/java/org/projectnessie/services/spi/TreeService.java

```diff
@@ -41,68 +46,126 @@ public interface TreeService {
+      @Valid
+          @NotNull
+          @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+          String refName,
+      Reference.ReferenceType type,
+      @Valid @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
+          String hash,
+      @Valid @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+          String sourceRefName)
       throws NessieNotFoundException, NessieConflictException;
 
   Reference assignReference(
       Reference.ReferenceType referenceType,
-      String referenceName,
-      String expectedHash,
-      Reference assignTo)
+      @Valid
+          @NotNull
+          @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+          String referenceName,
+      @Valid @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
+          String expectedHash,
+      @Valid Reference assignTo)
       throws NessieNotFoundException, NessieConflictException;
 
   void deleteReference(
-      Reference.ReferenceType referenceType, String referenceName, String expectedHash)
+      Reference.ReferenceType referenceType,
+      @Valid
+          @NotNull
+          @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+          String referenceName,
+      @Valid @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
+          String expectedHash)
       throws NessieConflictException, NessieNotFoundException;
 
   LogResponse getCommitLog(
-      String namedRef,
+      @Valid
+          @NotNull
+          @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+          String namedRef,
       FetchOption fetchOption,
-      String oldestHashLimit,
-      String youngestHash,
-      String filter,
-      Integer maxRecords,
-      String pageToken)
+      @Valid @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
+          String oldestHashLimit,
+      @Valid @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
+          String youngestHash,
+      @Nullable String filter,
+      @Nullable Integer maxRecords,
+      @Nullable String pageToken)
       throws NessieNotFoundException;
 
   MergeResponse transplantCommitsIntoBranch(
-      String branchName,
-      String expectedHash,
+      @Valid
+          @NotNull
+          @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+          String branchName,
+      @Valid @NotNull @Pattern(regexp = Validation.HASH_REGEX, message = Validation.HASH_MESSAGE)
+          String expectedHash,
       String message,
       List<String> hashesToTransplant,
-      String fromRefName,
+      @Valid
+          @NotBlank
+          @NotNull
+          @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
```

Comment:
a) does `REF_NAME_REGEX` handle the blanks too? Do we need this to be `NotBlank` or just `NotNull` is enough?

b) do we need validations for `hashesToTransplant` too?

Reply:
I used the same validation rules as in v1 `HttpTreeApi` and related classes.

`NotBlank` may actually produce a more intuitive error message,

Reply:
We could validate `hashesToTransplant`, but we do not do that in API v1 and did not want to change behaviour in this PR.

Reply:
I thought it is freshly added validations. Ok to be inline with v1 as of now and improve in the follow up.